2019-04-28  Leo Benjamin Gretzinger  <classlgr@lnxsrv07.seas.ucla.edu>

        * timespec.h (timespec_cmp): timespec_cmp now returns -1 if a is less than b,
        0 if they're equal and 1 otherwise.

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2019-04-28 03:19:14.716059000 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec < b.tv_sec)
+    return -1;
+  else if(a.tv_sce == b.tv_sec)
+    return 0;
+  else
+    return 1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
